### ISSUE TITLE
`cargo install s3s-fs --features binary` fails on Windows MSVC

### ISSUE THREAD

**Issue by amunra** [Nugine/s3s#52]
We use `s3s-fs`'s binary in our project to test our logic during unit testing.

Today I was setting up CIs across platforms and the Windows MSVC one failed to `cargo install s3s-fs --features binary`.

Digging a little bit, the cause of the issue is the following:

From: `crates/s3s-fs/Cargo.toml`
```toml
[features]
binary = ["tokio/full", "md-5/asm", "dep:clap", "dep:tracing-subscriber", "dep:hyper"]
                        ^______ here
```

Exposes the following bug: https://github.com/RustCrypto/hashes/issues/315

Which in turn manifests as the following build error:

```
PS C:\Users\Adam> cargo install s3s-fs --features binary
...
error: failed to run custom build command for `md5-asm v0.5.0`

Caused by:
  process didn't exit successfully: `C:\Users\Adam\AppData\Local\Temp\cargo-installbglGW6\release\build\md5-asm-82c27e24c0779d0f\build-script-build` (exit code: 1)
  --- stdout
...
  running: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.33.31629\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MD" "-O2" "-Brepro" "-W4" "-c" "-FoC:\\Users\\Adam\\AppData\\Local\\Temp\\cargo-installbglGW6\\release\\build\\md5-asm-5cb33a4a3b3b8b97\\out\\src/x64.o" "-c" "src/x64.S"
  cargo:warning=cl : Command line warning D9024 : unrecognized source file type 'src/x64.S', object file assumed
  cargo:warning=cl : Command line warning D9027 : source file 'src/x64.S' ignored
  cl : Command line warning D9021 : no action performed
...
  LINK : fatal error LNK1181: cannot open input file 'C:\Users\Adam\AppData\Local\Temp\cargo-installbglGW6\release\build\md5-asm-5cb33a4a3b3b8b97\out\src\x64.o'
  exit code: 1181
...
```

The good news is that disabling `md5-asm`, (i.e. `binary = ["tokio/full", "dep:clap", "dep:tracing-subscriber", "dep:hyper"]`) resolves the issue and it all builds just fine.

I couldn't really figure out a way to this in a way that is platform-specific: https://github.com/rust-lang/cargo/issues/1197

The only other way of getting this to work, as far as I know, is to split the binary from `s3s-fs` into its own separate crate `s3s-fs-bin` and change the install docs to:

```
cargo install s3s-fs-bin
```

I think that's pretty complicated though.

Would you take a quick PR to remove the `md-5/asm` feature?

Thanks!

**Comment by Nugine** [comment]
It's suprising that there is no warning in `md-5`. I think they should document this.

> Would you take a quick PR to remove the md-5/asm feature?

Welcome!

**Comment by amunra** [comment]
Yeah, they've got an open issue to add support for MSVC asm, but I don't understand why they just don't make the feature a no-op in the meantime. Doesn't matter too much I guess.

You'll be getting an PR from me soon :-)

**Comment by amunra** [comment]
See the PR. I've added Windows/MacOS CI too.

**Comment by amunra** [comment]
Assuming this gets merged, I'd be very very grateful for a new `0.6.0` release :-).
